main.dart

```diff
@@ -119,18 +123,22 @@ void main() {
         var newDescription = stdin.readLineSync()!;
         print("New Due Date (YYYY-MM-DD): ");
         var newDueDateStr = stdin.readLineSync()!;
-        var newDueDate = DateTime.parse(newDueDateStr);
-        print("Is it completed? (true or false): ");
-        var newStatusStr = stdin.readLineSync()!;
-        var newStatus = newStatusStr.toLowerCase() == 'true';
-        taskManager.editTask(
-          taskToEdit,
-          newTitle,
-          newDescription,
-          newDueDate,
-          newStatus,
-        );
-        print("Task updated successfully!\n");
+        try {
+          var newDueDate = DateTime.parse(newDueDateStr);
+          print("Is it completed? (true or false): ");
+          var newStatusStr = stdin.readLineSync()!;
+          var newStatus = newStatusStr.toLowerCase() == 'true';
+          taskManager.editTask(
+            taskToEdit,
+            newTitle,
+            newDescription,
+            newDueDate,
+            newStatus,
+          );
+          print("Task updated successfully!\n");
+        } catch (e) {
```

Comment:
Update the code to catch and handle specific error types.
Add a fallback option to handle other types of exceptions or unknown errors.